tests/bitExpert/PHPStan/Magento/Autoload/ExtensionInterfaceAutoloaderUnitTest.php

```diff
@@ -0,0 +1,137 @@
+            new Cache($cacheStorage),
+            $this->root->url()
+        );
+        $cacheStorage->expects(self::never())->method('load');
+        $autoloader->autoload('ExtensionInterfaceNotAtEnd');
+    }
+
+    /**
+     * @test
+     */
+    public function autoloadUsesCachedFileWhenFound(): void
+    {
+        $interfaceName = 'CachedExtensionInterface';
+        $cache = new Cache(new FileCacheStorage($this->root->url() . '/tmp/cache/PHPStan'));
+        $cache->save($interfaceName, '', "<?php interface ${interfaceName} {}");
+        $autoloader = new ExtensionInterfaceAutoloader($cache, $this->root->url());
+
+        static::assertFalse(interface_exists($interfaceName));
+        $autoloader->autoload($interfaceName);
+        static::assertTrue(interface_exists($interfaceName));
+    }
+
+    /**
+     * @test
+     */
+    public function autoloadDoesNotGenerateInterfaceWhenNoAttributesExist(): void
+    {
+        $interfaceName = 'NonExistentExtensionInterface';
+        $cache = new Cache(new FileCacheStorage($this->root->url() . '/tmp/cache/PHPStan'));
+        $autoloader = new ExtensionInterfaceAutoloader($cache, $this->root->url());
+
+        $autoloader->autoload($interfaceName);
+        static::assertFalse(interface_exists($interfaceName));
+    }
+
+    /**
+     * @test
+     */
+    public function autoloadGeneratesInterfaceWhenNotCached(): void
+    {
+        $interfaceName = 'UncachedExtensionInterface';
+        vfsStream::create([
+            'interface.php' => '<?php interface UncachedInterface {}',
+            'app' => [
+                'etc' => [
+                    'config.php' => <<<PHP
+<?php return [
+    'modules' => [
+        'Module_Name' => 1
+    ]
+];
+PHP
+                ],
+                'code' => [
+                    'Module_Name' => [
+                        'etc' => [
+                            'extension_attributes.xml' => <<<'XML'
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:framework:Api/etc/extension_attributes.xsd">
+    <extension_attributes for="UncachedInterface">
+        <attribute code="attr" type="string"/>
+    </extension_attributes>
+</config>
+XML
+                        ]
+                    ]
+                ]
+            ],
+        ], $this->root);
+
+        ComponentRegistrar::register(
+            ComponentRegistrar::MODULE,
+            'Module_Name',
+            $this->root->url() . '/app/code/Module_Name'
+        );
+
+        $autoloader = new ExtensionInterfaceAutoloader(
+            new Cache(new FileCacheStorage($this->root->url() . '/tmp/cache/PHPStan')),
+            $this->root->url()
+        );
+
+        require $this->root->url() . '/interface.php';
+
+        $autoloader->autoload($interfaceName);
+        static::assertTrue(interface_exists($interfaceName));
+        $interfaceReflection = new \ReflectionClass($interfaceName);
```

Comment:
I'm unsure how to resolve this error (outside of suppressing it). PHPStan definitely cannot find this class using static analysis!

Reply:
I am fine if we ignore the error. I mean me have to make some magic here, that's fine then :)